### PR TITLE
Add additional "can fail" and "timeout" fields to task groups

### PR DESCRIPTION
**Pull request by eramongodb**
Refer to the [Task Groups](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Project-Configuration-Files#task-groups) section of the "Project Configuration Files" page.

`evergreen validate` rejects a `*_timeout_secs` field of type `str`:

```yaml
task_groups:
  - name: example
    setup_group_timeout_secs: "123"
```
```
ERROR: load project error(s): loading file '[...]': error unmarshalling yaml strict: yaml: unmarshal errors:
  line 3: cannot unmarshal !!str `123` into int
```

The field types therefore updated from `Optional[Union[int, str]]` to `Optional[int]` accordingly.

### CODE REVIEW

File: src/shrub/v3/evg_task_group.py [hunk 1]
```diff
@@ -30,10 +34,15 @@ class EvgTaskGroup(BaseModel):
     max_hosts: Optional[int] = None
     share_processes: Optional[bool] = None
     setup_group_can_fail_task: Optional[bool] = None
-    setup_group_timeout_secs: Optional[Union[int, str]] = None
+    setup_group_timeout_secs: Optional[int] = None
     setup_group: Optional[List[EvgCommand]] = None
+    teardown_group_can_fail_task: Optional[bool] = None
```
Comment (namazgurbanovmdb) [inline src/shrub/v3/evg_task_group.py:39]: Can we add this to the class doc as well? Also, out of curiousity, is there timeout_specs for teardown_group?

Comment (eramongodb) [reply src/shrub/v3/evg_task_group.py]: > Can we add this to the class doc as well?

Apologies, `teardown_group_can_fail_task` is actually undocumented and invalid according to the EVG validator (which is a surprise to me). Filed [DEVPROD-12438](https://jira.mongodb.org/browse/DEVPROD-12032) and commented it out for now (similar approach to https://github.com/evergreen-ci/shrub.py/pull/40).

> Is there timeout_secs for teardown_group?

Yes, there is a `teardown_group_timeout_secs`. Added for completeness.

**Review by namazgurbanovmdb — APPROVED** [review]
LGTM

File: src/shrub/v3/evg_task_group.py (hunk 1 above)
Comment (eramongodb) [reply src/shrub/v3/evg_task_group.py]: [DEVPROD-12438](https://jira.mongodb.org/browse/DEVPROD-12438) clarified `teardown_group_can_fail_task` is deliberately unsupported due to implementation concerns and there are no plans to support it either. Removed the field entirely accordingly.

**Comment by eramongodb** [comment]
This PR recieved a LGTM several weeks ago but has not yet been merged. Is this blocked on anything?

**Pull request merged**